src/main/kotlin/org/wfanet/measurement/duchy/daemon/herald/Herald.kt

```diff
@@ -92,21 +104,23 @@ class Herald(
    * computations from the system computation service.
    */
   suspend fun syncStatuses(continuationToken: String): String {
-    var lastProcessedContinuationToken = continuationToken
     logger.info("Reading stream of active computations since $continuationToken.")
+
+    val streamRequest = streamActiveComputationsRequest {
+      this.continuationToken = continuationToken
+    }
+    var lastProcessedContinuationToken = continuationToken
     systemComputationsClient
-      .streamActiveComputations(
-        StreamActiveComputationsRequest.newBuilder().setContinuationToken(continuationToken).build()
-      )
+      .streamActiveComputations(streamRequest)
+      // TODO(@renjiez): Use new version of withRetriesOnEach() to add finally block to
+      //  failComputationAtKingdom
```

Comment:
New withRetriesOnEach() 
https://github.com/world-federation-of-advertisers/common-jvm/pull/161